From c46d606b778ad3d11f7ff7a744ed35219690a1fc Mon Sep 17 00:00:00 2001
From: Douglas Gregor <doug.gregor@gmail.com>
Date: Thu, 29 Apr 2010 00:54:26 +0000
Subject: [PATCH] Standards-conformance fix for Boost.Interprocess: max_value
 needs to be declared before it is used, since ADL won't
 look here.

[SVN r61667]
---
 .../containers/container/detail/utilities.hpp      |   16 ++++++++--------
 1 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/include/boost/interprocess/containers/container/detail/utilities.hpp b/include/boost/interprocess/containers/container/detail/utilities.hpp
index 6a0fd0e..1077938 100644
--- a/include/boost/interprocess/containers/container/detail/utilities.hpp
+++ b/include/boost/interprocess/containers/container/detail/utilities.hpp
@@ -24,6 +24,14 @@ namespace boost {
 namespace container {
 namespace containers_detail {
 
+template<class T>
+const T &max_value(const T &a, const T &b)
+{  return a > b ? a : b;   }
+
+template<class T>
+const T &min_value(const T &a, const T &b)
+{  return a < b ? a : b;   }
+
 template <class SizeType>
 SizeType
    get_next_capacity(const SizeType max_size
@@ -44,14 +52,6 @@ SizeType
    return max_size;
 }
 
-template<class T>
-const T &max_value(const T &a, const T &b)
-{  return a > b ? a : b;   }
-
-template<class T>
-const T &min_value(const T &a, const T &b)
-{  return a < b ? a : b;   }
-
 template<class SmartPtr>
 struct smart_ptr_type
 {
-- 
1.7.6.5

